### DISCUSSION TITLE
How can i detect objects on a video using an usb video grabber in realtime ?

### DISCUSSION THREAD

**Discussion by kev-steph** [ultralytics/yolov5#8396] [Q&A]
Hi everyone,
i am student.  I have a project for my master thesis und now i have a big problem.
I work with an ultrasonic device in the laboratory und my professor told me that i have to transfer the video from the ultrasonic device to my computer using an usb video grabber. That is not the problem. My problem is how can i detect objets on my video in real time on yolov 5 using this usb video grabber (the professor wants  this live detection with the usb video grabber).
 I don't know which source i have to use und how i have to program that(if i have to programm). I already trained the model on yolov 5 (but not in real time) und it works perfectly. 

With the webcam, i use :python detect .py --source 0, but i tried with source 1 ,2, 3 i have no result. 

I attach a video so that you can understand what i mean( this should be detected in real time)

If someone can help me, i will be very thankful.

[video_neu.zip](https://github.com/ultralytics/yolov5/files/9013749/video_neu.zip)

**Comment by glenn-jocher** [comment]
@kev-steph additional local cameras should have incremental indices, i.e. --source 1, etc. 

detect.py relies on cv2 for webcam interaction though, so before anything just make sure cv2 sees your webcam, i.e. see cv2 docs
https://www.geeksforgeeks.org/python-opencv-capture-video-from-camera/

**Reply by kev-steph** [reply]
@glenn-jocher Ok , if i good understand , do i have to do that with this video grabber ? 
With my webcam , i don’t have a problem . For this project ,my webcam is this usb video grabber . One more thing: my computer doesn’t show the usb grabber in the C directory, is that normal?

**Reply by kev-steph** [reply]
@glenn-jocher it (the usb grabber)works with a software that i installed on my computer .

**Reply by glenn-jocher** [reply]
@kev-steph well if your attached camera is the only camera on your system then the index should be 0, but again test with cv2 first to make sure it's found and reading images.

```
python detect.py --source 0
```

**Reply by kev-steph** [reply]
@glenn-jocher may be you didn’t understand what I mean. My integrated camera works perfectly. I want to detect objects with the usb video grabber (my professor wants that)

**Reply by kev-steph** [reply]
@glenn-jocher detection in real time with this usb video grabber . I wrote that above

**Comment by kev-steph** [comment]
@glenn-jocher the  usb grabber works with this software on the photo below.
![78439D7B-26C9-4BD6-B8E2-EA8D39AED664](https://user-images.githubusercontent.com/108429915/176714574-a8e6f24a-4e0d-4c64-af30-0e3e0c38a677.jpeg)

**Reply by glenn-jocher** [reply]
@kev-steph I'm sorry we don't have resources to help debug local environments, I would talk to your IT administrator or professor about local environment/hardware issues.

**Reply by kev-steph** [reply]
Ok I will talk with him again about that .  
That seems to be very difficult . Thanks
